contracts/predicates/GOFPPredicates.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * Authors: Moonstream Engineering (engineering@moonstream.to)
+ * GitHub: https://github.com/bugout-dev/engine
+ */
+
+pragma solidity ^0.8.9;
+
+import {IERC721} from "@openzeppelin/contracts/contracts/token/ERC721/IERC721.sol";
+import {IGOFP} from "../interfaces/IGOFP.sol";
+
+library GOFPPredicates {
+    function doesNotExceedMaxTokensInSession(
+        uint256 maxStakable,
+        address gofpAddress,
+        uint256 sessionId,
+        address player,
```

Comment:
@kellan-simiotics : Can you create a .md file in this PR which discusses predicates, and explains our reasoning behind the signatures? We can pull content from our slack conversation about it.